### DISCUSSION TITLE
Offline stern installer

### DISCUSSION THREAD

**Discussion by asheeshv** [stern/stern#266] [General]
Hello there,

I've been using stern for about 6 months, and I find it a wonderful debugging tool with the ability to tail logs from multiple containers. I've been mainly using stern on my local dev box and I install it using choco. However, our production setup is an offline k8s cluster which doesn't have access to the internet or outside world. It's a closed system. I wanted to know if there's a way to offline install stern or carry the necessary setup which I can untar and it runs seamlessly.
I tried copying the stern_1.25.0_linux_amd64.tar.gz on my Ubuntu VM running microk8s, and untarred, but it didn't execute. May be I'm missing something here, and need a dependency that needs to be installed as well. Any help would be appreciated!
Thanks,
Asheesh

**Comment by tksm** [comment]
Hi,

Stern does not have any dependencies, so you can use it by copying a single binary of `stern` from `stern_1.25.0_linux_amd64.tar.gz`.

The following steps worked for me.

```bash
$ wget https://github.com/stern/stern/releases/download/v1.25.0/stern_1.25.0_linux_amd64.tar.gz
$ tar zxvf stern_1.25.0_linux_amd64.tar.gz

$ sudo mv stern /usr/local/bin/
$ stern --version
version: 1.25.0
commit: f13bde422e977c7a69ec0827c0337b2bc8e44444
built at: 2023-04-13T23:06:41Z

# stern does not have any dependencies
$ file /usr/local/bin/stern  | grep "statically linked"
/usr/local/bin/stern: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=mJiKgS85P1aqZA1MKfYp/OttM8NMvHsVFQ8p6r0Qa/JgQSQfui7eNo34dC0y3F/i7qEzHGUHaqrR8z67qmw, stripped
```

**Reply by asheeshv** [reply]
Thank you so much Takashi. This one worked just fine!
